fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -56,9 +55,22 @@ class WebExtensionPromptFeature(
             flow.mapNotNull { state ->
                 state.webExtensionPromptRequest
             }.distinctUntilChanged().collect { promptRequest ->
-                val addon = provideAddons().find { addon ->
-                    addon.id == promptRequest.extension.id
+                // The install flow in Fenix relies on an [Addon] object so let's convert the (GeckoView)
+                // extension into a minimal add-on. The missing metadata will be fetched when the user
+                // opens the add-ons manager.
+                val addon = promptRequest.extension.getMetadata()?.let { m ->
+                    Addon(
+                        id = promptRequest.extension.id,
+                        downloadUrl = promptRequest.extension.url,
+                        version = m.version,
+                        permissions = m.permissions + m.hostPermissions,
+                        translatableName = mapOf(Addon.DEFAULT_LOCALE to m.name.orEmpty()),
+                        translatableDescription = mapOf(Addon.DEFAULT_LOCALE to m.description.orEmpty()),
+                        // We don't have a summary at this point so let's reuse the description.
+                        translatableSummary = mapOf(Addon.DEFAULT_LOCALE to m.description.orEmpty()),
+                    )
                 }
```

Comment:
Nice!!

Reply:
We will need to test by changing a different locale from the default one and see how it plays. I'm not sure if we may loose translations in permission popup?

Reply:
yeah, this is still a WIP.  I guess writing some tests might help us uncover bugs (I hope so..)